lms/djangoapps/edraak_marketing_email/helpers.py

```diff
@@ -0,0 +1,17 @@
+from edraak_marketing_email.models import UnsubscribedUser
+
+
+def unsubscribe_from_marketing_emails(user):
+    is_unsubscribed = UnsubscribedUser.objects.filter(user=user).exists()
+    if is_unsubscribed:
+        return
+
+    UnsubscribedUser.objects.create(user=user)
+
+
+def subscribe_to_marketing_emails(user):
+    is_unsubscribed = UnsubscribedUser.objects.filter(user=user).exists()
+    if not is_unsubscribed:
+        return
+
+    UnsubscribedUser.objects.filter(user=user).delete()
```

Comment:
Same here `filter(...).delete()` works okay if not object was found -- it doesn't do anything in such cases.

```suggestion
def subscribe_to_marketing_emails(user):
    UnsubscribedUser.objects.filter(user=user).delete()
```